libs/payments/stripe/src/lib/stripe.util.ts

```diff
@@ -95,3 +96,31 @@ export const getSubscribedProductIds = (subscribedPlans: StripePlan[]) => {
     ? subscribedPlans.map(({ product: productId }) => productId as string)
     : [];
 };
+
+/**
+ * Formats a Stripe Invoice to the FirstInvoicePreview DTO format.
+ */
+export function stripeInvoiceToFirstInvoicePreviewDTO(
+  invoice: StripeUpcomingInvoice
+) {
+  const taxAmounts = [
+    {
+      title: invoice.total_tax_amounts[0].tax_rate.display_name,
+      inclusive: invoice.total_tax_amounts[0].inclusive,
+      amount: invoice.total_tax_amounts[0].amount,
+    },
+  ];
+
+  const discountAmount =
+    invoice.total_discount_amounts && invoice.total_discount_amounts.length > 0
+      ? invoice.total_discount_amounts[0].amount
+      : null;
```

Comment:
If you want a visually unpleasant one-liner, optional chaining w/ nullish coalescing to the "rescue":
```js
const discountAmount = invoice.total_discount_amounts?.[0].amount ?? null;
```